### ISSUE TITLE
Question: Instructions on which files to replace in the official jellyfin (.deb)

### ISSUE THREAD

**Issue by jhonatanDS** [flathub/org.jellyfin.JellyfinServer#397]
Currently, I am using version 10.8.13 of Jellyfin Flatpak, but my smart TVs are requiring version 10.9. Since there is no forecast for releasing version 10.9 in Flatpak, I would like to know if you could help us with the following:

1. Providing a script that installs the official version of Jellyfin and replaces the files (database) with those from the Flatpak version, importing the data from the Flatpak version;
2. Or providing instructions on which files and directories we should manually replace in order to keep our data in version 10.9.

We will probably need to download version 10.8.13 in .deb format (or another), install it, replace the files with the ones from Flatpak, and then upgrade again using the .deb (or another format) to version 10.9.

I just don't want to lose the information about which content/episodes have already been watched and the synopses that I personally wrote/translated.

**Comment by lwbt** [comment]
- The flatpak version stores data in `$HOME/.var/app/org.jellyfin.JellyfinServer/data/jellyfin`.
- The Debian packages seem to store data in  `$HOME/.local/share/jellyfin`.
https://jellyfin.org/docs/general/administration/configuration/#data-directory

Stop the flatpak version from the dashboard. Install the Debian packages on your system by following the official documentation. Create a simple configuration and eventually put a few files to scan in an example library. Stop the Debian installation from the dashboard. This should give you structure to compare against. Make a copy of both folders. Move the current configuration for the Debian version aside to make room for the data you are going to migrate from flatpak to Debian. The official release notes always recommend to backup the configuration by simply making a copy of the folder. So it shouldn't be too difficult. Ideally the folder structure is the same or parts which don't fit will be ignored or recreated.

The only thing from my point of view that is stopping you from running both versions side by side is the port number.

The dashboard is actually not that complicated. Depending on how many users you have, it may be easier to start from scratch with the latest version of Jellyfin. Your could use an extension like https://github.com/gildas-lormeau/SingleFile to document every important page and setting, then set everything up as your need it and check documentation if settings are still relevant. Finally set up and scan libraries.

Apparently someone wrote a script to migrate from Windows to Linux, but it doesn't seem to be maintained currently. https://jellyfin.org/docs/general/administration/migrate

**Comment by lwbt** [comment]
> Since there is no forecast for releasing version 10.9 in Flatpak

I took over maintainership a few days ago and published v10.10.

> I would like to know if you could help us with the following:
>
> 1. Providing a script that installs the official version of Jellyfin ...

https://jellyfin.org/downloads/server -> `curl -s https://repo.jellyfin.org/install-debuntu.sh | sudo bash`

> ... and replaces the files (database) with those from the Flatpak version, importing the data from the Flatpak version;

Perfectly scripted migrations will require considerable upfront work and cannot be done at the moment, probably never.

As I said before, users (administrators) of Jellyfin should document their settings to know how they configured it and where media is located and how it is backed up. Regenerating data should be easier when migrating from one solution to another and avoids cruft. Most of the plain text configuration files can be compared and modified by hand. Databases like the smaller core database need to be investigated but might create a huge maintenance burden over time with changing database schemes. Changed file paths in the library database could be modified with a query, but ideally one should use the same location with both packaging formats when keeping metadata is desired and longer scan an regeneration want to be avoided.

> Or providing instructions on which files and directories we should manually replace in order to keep our data in version 10.9.

That is indeed something I see which many users will need and eventually will give the Flatpak version a bad reputation when users have trouble moving their data and configuration from one system with one type of packaging to a different one.

I created a first draft here to address some of the concerns: https://github.com/flathub/org.jellyfin.JellyfinServer/wiki/Data-Migration

Thank you for bringing this up.
